ast.go

```diff
@@ -28,13 +28,23 @@ import (
 type Serializer interface {
 	// Serialize writes the statement/expression to the io.Writer. If an
 	// error is returned the io.Writer may contain partial output.
-	Serialize(w io.Writer) error
+	Serialize(w io.Writer, interpolate bool) error
```

Comment:
I don't think `interpolate` is the right term. How about `placeholders`?

Adding the `interpolate` argument to all of the `Serialize` methods is sort of ugly, though it does have the advantage of being obvious. The alternative that comes to mind is to create some sort of specialized `io.Writer` interface. Something along the lines of:

```
type sqlWriter interface {
  io.Writer
  WriteBytes(b BytesVal) error
  WriteEncoded(e encodedVal) error
  WriteNum(n NumVal) error
  WriteRaw(r RawVal) error
  WriteString(s StrVal) error
}
```

You'd then provide different implementations for serializing as normal vs serializing-with-placeholders:

```
type standardWriter struct {
  bytes.Buffer
}

func (w *standardWriter) WriteBytes(b BytesVal) error {
  return encodeSQLBytes(w, []byte(b))
}

...

type placeholderWriter struct {
  bytes.Buffer
}

func (w *placeholderWriter) WriteBytes(b BytesVal) error {
   _, err := w.Write(astPlaceholder)
  return err
}
```

This is just some brainstorming. I'm not sure if the above will work, but it seems like it should.